packages/backend/src/models/DashboardModel/DashboardModel.ts

```diff
@@ -876,4 +879,180 @@ export class DashboardModel {
             .distinctOn(`${DashboardVersionsTableName}.dashboard_id`)
             .where(`${ProjectTableName}.project_uuid`, projectUuid);
     }
+
+    private async checkDashboardTileExistsInDashboard(
+        dashboardUuid: string,
+        dashboardTileUuid: string,
+    ) {
+        // NOTE: ensure that this dashboard actually contains the tile, since tile uuids might not be unique across dashboards
+        const dashboardTile = await this.database(DashboardTilesTableName)
+            .join(
+                DashboardVersionsTableName,
+                `${DashboardVersionsTableName}.dashboard_version_id`,
+                '=',
+                `${DashboardTilesTableName}.dashboard_version_id`,
+            )
+            .join(
+                DashboardsTableName,
+                `${DashboardsTableName}.dashboard_id`,
+                '=',
+                `${DashboardVersionsTableName}.dashboard_id`,
+            )
+            .where(`${DashboardsTableName}.dashboard_uuid`, dashboardUuid)
+            .andWhere(
+                `${DashboardTilesTableName}.dashboard_tile_uuid`,
+                dashboardTileUuid,
+            )
+            .select(`${DashboardTilesTableName}.dashboard_tile_uuid`)
+            .first();
+
+        if (!dashboardTile) {
+            throw new NotFoundError('Dashboard tile not found for dashboard');
+        }
+    }
+
+    async createComment(
+        dashboardUuid: string,
+        dashboardTileUuid: string,
+        text: string,
+        replyTo: string | null,
+        user: LightdashUser,
+    ): Promise<string> {
+        await this.checkDashboardTileExistsInDashboard(
+            dashboardUuid,
+            dashboardTileUuid,
+        );
+        const [{ comment_id: commentId }] = await this.database(
+            DashboardTileCommentsTableName,
+        )
+            .insert({
+                text,
+                dashboard_tile_uuid: dashboardTileUuid,
+                reply_to: replyTo ?? null,
+                user_uuid: user.userUuid,
+            })
+            .returning('comment_id');
+
+        return commentId;
+    }
+
+    async findCommentsForDashboard(
+        dashboardUuid: string,
+        userUuid: string,
+        canUserRemoveAnyComment: boolean,
+    ): Promise<Record<string, Comment[]>> {
+        const dashboard = await this.getById(dashboardUuid);
+
+        const tileUuids = dashboard.tiles.map((tile) => tile.uuid);
+
+        const commentsWithUsers = await this.database(
```

Comment:
we're missing some typing here,  because it thinks commentsWithUsers is any[]

Reply:
I think you didn't add `dbComments` to knex-tables.d.ts

Reply:
I didn't you're right... just did!

Reply:
this improves types: https://github.com/lightdash/lightdash/pull/9018/commits/226e51f0d94c9581f4e874cdfc3aa3f8e5d84340